### PR TITLE
Feature/25 merge gitlab

### PR DESCRIPTION
**Pull request by ErikBrendel**

### CODE REVIEW

File: js/frontend.js [hunk 1]
```diff
@@ -847,7 +847,7 @@ const frontend = {
         // if shownSemesters has not been initialized so far ..
         if (semesterManager.shownSemesters.length === 0) {
             // .. initialize starting at semesterManager.firstSemester-Semester
```
Comment (MerlindlH) [inline js/frontend.js]: i have the feeling this has to be updated

**Review by Markus-Brand — APPROVED** [review]

File: js/logic.js [hunk 2]
```diff
@@ -54,34 +67,95 @@ const semesterManager = {
...
-    lastSummerSemester: "SS17",
-    lastWinterSemester: "WS17/18",
-    preLastSummerSemester: "SS16",
-    preLastWinterSemester: "WS16/17",
-    /* the semester that is the first semester when you first start the application */
-    firstSemester: "WS15/16",
+    currentSemester: function () {
+        var currentDate = getCurrentDate();
+
+        if(currentDate.month < 4) {
```
Comment (MerlindlH) [inline js/logic.js:98]: is this month zero or one-indexed?

Comment (MerlindlH) [reply js/logic.js:98]: maybe this could be extracted

**Review by MerlindlH — COMMENTED** [review]
pew

File: js/logic.js (hunk 2 above)
Comment (ErikBrendel) [reply js/logic.js:98]: one

Comment (ErikBrendel) [reply js/logic.js:98]: no

**Pull request merged**